### PR TITLE
Fix market bootstrap reliability and scan performance

### PR DESCRIPTION
**Pull request by xang1234**
## Summary

Fixes several market bootstrap blockers and speeds up the initial bootstrap scan path.

## Changes

- Add runtime fixes for KR/CN universe refresh failures, including pykrx import support, KR listing fallback behavior, bounded CN universe fetch duration, and CN listing fallback handling.
- Improve bootstrap status handling so optional/partial failures do not incorrectly block unrelated market work.
- Reduce initial bootstrap scan cost by excluding `setup_engine` from cache-gated bootstrap defaults while preserving full manual/daily scan defaults.
- Avoid nested setup-engine detector thread pools by making detector execution serial by default, with explicit detector concurrency still available for diagnostics.
- Add INFO-level slow setup-engine detector summaries for future scan performance diagnosis.
- Add tests covering bootstrap profile boundaries and setup-engine detector execution behavior.

## Validation

- `backend/venv/bin/pytest backend/tests/unit/test_aggregator_execution_pipeline.py backend/tests/unit/test_scan_defaults.py backend/tests/unit/test_feature_store_tasks.py backend/tests/unit/test_setup_engine_screener.py -q`: `74 passed`
- `backend/venv/bin/pytest backend/tests/unit/test_cn_market_data_service.py backend/tests/unit/test_universe_tasks.py backend/tests/unit/test_official_market_universe_source_service.py -q`: `39 passed`
- `backend/venv/bin/pytest backend/tests/unit/test_cn_market_data_service.py backend/tests/unit/test_provider_circuit_breaker.py backend/tests/unit/test_celery_config.py -q`: previously passed during this branch work
- `git diff --check`
- `python -m compileall` for changed app files

## Notes

- `bd` is not installed in this local shell, so bead sync commands could not run.
- Local working tree still has unrelated `.beads`, `.plans`, and telemetry files that were not included in this PR.

## Summary by CodeRabbit

* **New Features**
  * Parallel detector execution and bounded parallel scanning for faster pattern scans
  * Lightweight bootstrap scan profile that omits heavy screening
  * Progress updates and enforced time limits for market-universe refreshes
  * INFO-level slow-detector timing summaries in logs
  * CI workflows expanded to include Korea (KR) market

* **Bug Fixes**
  * Timeout protection and deterministic fallback for market-data fetching (CN/KR)
  * AKShare retry/cooldown and fallback to alternate sources for OHLCV
  * Safer persistence path for FX rates with atomic upsert on supported DBs

* **Tests**
  * Expanded unit and performance tests covering parallel detectors, timeouts, fallbacks, and bootstrap profiles

* **Chores**
  * Runtime dependency pin added to preserve compatibility (setuptools<81)

### CODE REVIEW

**Comment by xang1234** [comment]
Addressed the review feedback in `54087cd7c`:

- `backend/app/services/kr_market_data_service.py`: validates the raw `short_code` before zero-padding so blank listing rows are skipped instead of becoming `000000.KS`; added a regression row in `test_kr_market_data_service.py`.
- `backend/tests/unit/use_cases/test_run_bulk_scan.py`: made the threaded scan assertion order-insensitive.
- `backend/tests/performance/test_setup_engine_performance.py`: restored a strict serial trace timing bound and added a separate concurrent trace timing test for worker overlap.
- `backend/app/analysis/patterns/high_tight_flag.py`: added `strict=True` to the pole-window `zip`.
- `backend/app/analysis/patterns/aggregator.py`: documented the frozen/read-only detector input contract instead of copying large OHLCV frames in the diagnostic threaded path.

Validation run:

```bash
backend/venv/bin/pytest \
  backend/tests/unit/test_kr_market_data_service.py \
  backend/tests/unit/use_cases/test_run_bulk_scan.py \
  backend/tests/unit/test_aggregator_execution_pipeline.py \
  backend/tests/unit/test_detector_subtasks_c3a_c4a_c6a.py::test_high_tight_flag_returns_flag_validated_candidate \
  backend/tests/unit/test_detector_subtasks_c3a_c4a_c6a.py::test_high_tight_flag_reports_rejection_reason_when_flag_missing \
  backend/tests/unit/test_detector_subtasks_c3a_c4a_c6a.py::test_high_tight_flag_validates_beyond_first_five_poles \
  backend/tests/unit/test_detector_subtasks_c3a_c4a_c6a.py::test_high_tight_flag_rejects_invalid_volume_baseline \
  backend/tests/performance/test_setup_engine_performance.py::TestAggregatorBudget::test_serial_trace_timing_sum_tracks_wall_clock \
  backend/tests/performance/test_setup_engine_performance.py::TestAggregatorBudget::test_concurrent_trace_timing_sum_allows_worker_overlap \
  -q
```

Result: `50 passed, 2 warnings in 8.80s`.

Also passed: `git diff --check` and `compileall` for the changed backend modules.

Note: the full `backend/tests/performance/test_setup_engine_performance.py` file still has an existing `three_weeks_tight` detector budget failure on this machine (`2170.7ms` on rerun vs `300ms`), unrelated to this review change.

File: backend/app/services/kr_market_data_service.py [hunk 1]
```diff
@@ -113,7 +128,20 @@ def listing_rows(
...
+                logger.warning(
+                    "KRX daily ticker fetch failed for %s on %s: %s",
+                    normalized_board,
+                    as_of_token,
+                    exc,
+                )
+                tickers = []
+            if not tickers:
+                rows.extend(self._current_listing_rows(normalized_board))
+                continue
```
[the remark this answers is not in the document]
Comment (xang1234) [reply backend/app/services/kr_market_data_service.py:150]: Addressed in `backend/app/services/kr_market_data_service.py`: current-listing fallback now only runs for `as_of=None` or today, and historical empty ticker responses are skipped to preserve point-in-time correctness. Added regression coverage in `backend/tests/unit/test_kr_market_data_service.py`.

File: backend/app/services/cn_market_data_service.py [hunk 2]
```diff
@@ -340,24 +433,44 @@ def daily_ohlcv(
...
+                if rows:
+                    self._record_akshare_ohlcv_success()
+                    return rows
+            except CnDependencyError:
+                raise
+            except Exception as exc:  # pragma: no cover - network variability
+                self._record_akshare_ohlcv_failure()
+                logger.warning("AKShare CN OHLCV fetch failed for %s: %s", code, exc)
 
         return self._daily_ohlcv_from_baostock(code, start=start_token, end=end_token)
```
[the remark this answers is not in the document]
Comment (xang1234) [reply backend/app/services/cn_market_data_service.py:459]: Addressed in `backend/app/services/cn_market_data_service.py`: AKShare OHLCV fetches now run through `_call_with_timeout`, so timeouts can be recorded and BaoStock fallback can run. Added regression coverage in `backend/tests/unit/test_cn_market_data_service.py`.

File: backend/tests/unit/test_cn_market_data_service.py [hunk 3]
```diff
@@ -92,6 +95,122 @@ def stock_zh_a_spot_em():
...
+        @staticmethod
+        def stock_info_a_code_name():
+            return pd.DataFrame([{"code": "000001", "name": "平安银行"}])
+
+    service = CnMarketDataService(akshare_module=FallbackAkshare(), timeout_seconds=1)
+    started_at = time.monotonic()
+
+    rows = service.listing_rows(as_of=date(2026, 4, 30))
+
+    assert time.monotonic() - started_at < 3
```
[the remark this answers is not in the document]
Comment (xang1234) [reply backend/tests/unit/test_cn_market_data_service.py:164]: Addressed in `backend/tests/unit/test_cn_market_data_service.py`: the SIGALRM-dependent timeout tests now use `pytest.mark.skipif` when signal timers are unavailable, matching `_call_with_timeout` platform behavior.
